pepper-apis/dsm-core/src/test/resources/elastic/lmsMappings.json

```diff
@@ -0,0 +1,3812 @@
+{
```

Comment:
How is this mapping generated?

Reply:
I ran this ES query: ` curl -u pepper_engineering:'xxxxxx' -XGET --header 'Content-Type: application/json' https://f7df559584f84f2cab27de822618fb89.us-central1.gcp.cloud.es.io:9243/participants_structured.cmi.cmi-lms/_mapping`.

Refreshing the mapping from an ES instance at test time generates a dependency on an external resource, which could be a problem. Also, this is just the mapping I chose for these tests, so other tests may prefer other mappings.

You could imagine a scheduled job that updates the test resources from an ES instance, to catch any mapping changes, but I wonder it the mapping update process is better done deliberately by hand?

Pulling back and thinking through scenarios... First, how often do ES mappings change? (DSM code updates mappings on the fly, but these are limited to dynamic fields.) If we make a deliberate mapping change, would if make sense to first verify the change with a test using the ES test containers? 

Is it better to deal with broken tests from automatically updating mappings, or is it better to have tests break because their mapping is out of date with DSM ES code under test? If we do not change mappings very often, does it make more sense to spend our development time increasing test coverage of DSM ES code, or automating test mapping update?

Reply:
ES mappings will not change much from here on out.  I think it's better to have tests break because their mapping is out of date with ES code under test.  I think we need to add the curl snippet and a few sentences about the new ES testing infrastructure to our internal dev documentation so that when people do go to change the mappings, they are leverage this new infrastructure for testing.  @denniscunningham @ssettipalli  can you drop some updates in those docs?

Reply:
I can update those docs, but I am not sure where they are. @ssettipalli, what do you suggest?